Source/TrieRouteMatcher.swift

```diff
@@ -35,36 +35,42 @@ public struct TrieRouteMatcher: RouteMatcherType {
         var nextComponentId = 1
 
         for route in routes {
-
-            // turn component (string) into an id (integer) for fast comparisons
-            let componentIds = route.path.split("/").map { component -> Int in
-
-                // if it already has a component with the same name, use that id
-                if let id = componentsTrie.findPayload(component.characters) {
+            
+            for method in route.methods {
+                
+                // add the method to the path so it is checked
+                let path = method.description + route.path
```

Comment:
This should be `method.description + "/" + route.path`. As is, the string turns into something like `GETmy/path`. It should be something like `GET/my/path` so that it is properly distinguished as a component on the matching side.